state-chain/chains/src/eth.rs

```diff
@@ -77,6 +77,43 @@ impl EthereumTrackedData {
 	}
 }
 
+mod fees {
+	// TODO: refine these constants.
+	pub const BASE_COST_PER_BATCH: u128 = 10_000;
+	pub const GAS_COST_PER_FETCH: u128 = 40_000;
+	pub const GAS_COST_PER_TRANSFER: u128 = 50_000;
+}
+
+impl FeeEstimationApi<Ethereum> for EthereumTrackedData {
+	fn estimate_ingress_fee(
+		&self,
+		asset: <Ethereum as Chain>::ChainAsset,
+	) -> <Ethereum as Chain>::ChainAmount {
+		use fees::*;
+
+		// Note: this is taking the egress cost of the swap in the ingress currency (and basing the
+		// cost on the ingress chain).
+		let gas_cost_per_fetch = BASE_COST_PER_BATCH +
+			match asset {
+				assets::eth::Asset::Eth => Zero::zero(),
+				assets::eth::Asset::Flip | assets::eth::Asset::Usdc => GAS_COST_PER_FETCH,
+			};
+
+		(self.base_fee + self.priority_fee).saturating_mul(gas_cost_per_fetch)
+	}
+
+	fn estimate_egress_fee(
+		&self,
+		_asset: <Ethereum as Chain>::ChainAsset,
+	) -> <Ethereum as Chain>::ChainAmount {
+		use fees::*;
+
+		let gas_cost_per_transfer = BASE_COST_PER_BATCH + GAS_COST_PER_TRANSFER;
```

Comment:
I'd suggest to differentiate on the egress between tokens and ETH. While it might be a bit of micromanaging, it is just a small code change and the difference is like 20k gas, so it's not small.